--- ../smack_src_3_1_0/source/org/jivesoftware/smack/sasl/SASLMechanism.java	2008-11-21 00:31:48.000000000 -0500
+++ SASLMechanism.java	2008-12-23 22:17:06.875000000 -0500
@@ -149,10 +149,13 @@
             response = sc.evaluateChallenge(null);
         }
 
-        String authenticationText = Base64.encodeBytes(response,Base64.DONT_BREAK_LINES);
-        if(authenticationText.equals("")) {
-            authenticationText = "=";
-        }
+		String authenticationText = null;
+		if(null != response) {
+			authenticationText = Base64.encodeBytes(response,Base64.DONT_BREAK_LINES);
+		}
+		if((null == authenticationText) || (authenticationText.equals(""))) {
+			authenticationText = "=";
+		}
 
         stanza.append("<response xmlns=\"urn:ietf:params:xml:ns:xmpp-sasl\">");
         stanza.append(authenticationText);
